src/System.IO.Pipelines/tests/StreamPipeWriterTests.cs

```diff
@@ -61,6 +61,25 @@ public async Task DataFlushedOnCompleteAsync()
             Assert.Equal("Hello World", Encoding.ASCII.GetString(stream.ToArray()));
         }
 
+        [Fact]
+        public async Task CompleteAsyncDoesNotThrowObjectDisposedException()
```

Comment:
Fix complete as well.

Reply:
I'll reorder in complete, but it doesn't have this problem as the cts isn't used.